src/main/java/seedu/address/logic/parser/ParseArgument.java

```diff
@@ -0,0 +1,108 @@
+package seedu.address.logic.parser;
+
+import static java.util.Objects.requireNonNull;
+
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import seedu.address.logic.parser.exceptions.ParseException;
+import seedu.address.model.lecture.LectureName;
+import seedu.address.model.module.ModuleCode;
+import seedu.address.model.tag.Tag;
+import seedu.address.model.video.VideoName;
+
+
+/**
+ * A dummy class containing utility methods used for parsing strings into ModuleCode, LectureName, and Video.
+ */
+public class ParseArgument {
+
+    /**
+     * Parses a {@code String moduleCode} into a {@code ModuleCode}.
+     * Leading and trailing whitespaces will be trimmed.
+     *
+     * @throws ParseException if the given {@code name} is invalid.
+     */
+    public static ModuleCode parseModule(String moduleCode) throws ParseException {
+        requireNonNull(moduleCode);
+        String trimmedModuleCode = moduleCode.trim();
+        if (!ModuleCode.isValidCode(trimmedModuleCode)) {
+            throw new ParseException(ModuleCode.MESSAGE_CONSTRAINTS);
+        }
+        return new ModuleCode(moduleCode);
+    }
```

Comment:
Seems to be duplicated code, could you use `ParseUtil.parseModuleCode(String)`?

As much as possible, don't create dummy `Module` objects just for the sake of retrieving the real `Module` object. `Model` has a method for retrieving `Module` using `ModuleCode`.